### PR TITLE
[extension/encoding/otlp_encoding] add a new extension for OTLP encoding

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
This creates a new extension for OTLP encoding, so we can use OTLP encoding using the mechanism designed in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28686

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28686

**Testing:**
100 % test coverage

**Documentation:**
Very basic README.

### CODE REVIEW

File: extension/encoding/otlpencodingextension/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+type: otlpencoding
```
Comment (dmitryax) [inline extension/encoding/otlpencodingextension/metadata.yaml:1]: Let's call the encoding extension this way? Similar to the storage extensions e.g. file_storage

```suggestion
type: otlp_encoding
```

Comment (atoulme) [reply extension/encoding/otlpencodingextension/metadata.yaml]: Changed.

Comment (dmitryax) [reply extension/encoding/otlpencodingextension/metadata.yaml]: Sorry, I meant the component type name only. The go module should not have underscores

**Comment by atoulme** [comment]
Not sure if this one should go in or https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27564.

**Comment by VihasMakwana** [comment]
@atoulme I think we can close this in favor of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27564, as latter has support for `proto` as well.

**Comment by atoulme** [comment]
Makes sense. Closing.

**Pull request closed without merging**